### DISCUSSION TITLE
Compilation error on arm64 platform

### DISCUSSION THREAD

**Discussion by imChipo** [dbeaver/dbeaver#9091] [DBeaver Development] [answered]
**Is your feature request related to a problem? Please describe.**

By default, Dbeaver compiles different versions for windows, linux and MacOS.  They are all based on x86_64 architecture.  When I add aarch64 architecture support to the file pom.xml, the compilation will report an error.

[INFO] org.jkiss.dbeaver.ext.office.feature 1.1.72-SNAPSHOT SUCCESS [  0.062 s]
[INFO] org.jkiss.dbeaver.ext.ui.svg.feature 1.0.71-SNAPSHOT SUCCESS [  0.058 s]
[INFO] features 1.0.0-SNAPSHOT ............................ SUCCESS [  0.005 s]
[INFO] org.jkiss.dbeaver.ext.test 1.0.139-SNAPSHOT ........ SUCCESS [  0.037 s]
[INFO] tests 1.0.0-SNAPSHOT ............................... SUCCESS [  0.005 s]
[INFO] DBeaver CE AppStore 7.1.1-SNAPSHOT ................. FAILURE [ 13.705 s]
[INFO] DBeaver CE Standalone 7.1.1-SNAPSHOT ............... SKIPPED
[INFO] DBeaver P2 Repositories 1.0.0-SNAPSHOT ............. SKIPPED
[INFO] DBeaver CE Repository 1.0.0-SNAPSHOT ............... SKIPPED
[INFO] DBeaver - Debug support 1.0.0-SNAPSHOT ............. SKIPPED
[INFO] DBeaver - Office support 1.0.0-SNAPSHOT ............ SKIPPED
[INFO] DBeaver - SVG support 1.0.0-SNAPSHOT ............... SKIPPED
[INFO] DBeaver - SSHJ support 1.0.0-SNAPSHOT .............. SKIPPED
[INFO] DBeaver - Git support 1.0.0-SNAPSHOT ............... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:26 min
[INFO] Finished at: 2020-06-24T10:41:58+08:00
[INFO] ------------------------------------------------------------------------

[ERROR] Failed to execute goal org.eclipse.tycho:tycho-p2-director-plugin:1.7.0:materialize-products (materialize-products) on project appstore: Installation of product org.jkiss.dbeaver.core.appstore for environment linux/gtk/aarch64 failed: Call to p2 director application failed with exit code 13. Program arguments were: [-metadataRepository, file:/home/lio/dvp/dbeaver/product/appstore/target/,file:/home/lio/dvp/dbeaver/product/appstore/target/targetPlatformRepository/, -artifactRepository, file:/home/lio/dvp/dbeaver/product/appstore/target/,file:/resolution-context-artifacts@%252Fhome%252Flio%252Fdvp%252Fdbeaver%252Fproduct%252Fappstore,file:/home/lio/dvp/dbeaver/features/org.jkiss.dbeaver.appstore.feature/target/,file:/home/lio/dvp/dbeaver/features/org.jkiss.dbeaver.rcp.feature/target/,file:/home/lio/dvp/dbeaver/features/org.jkiss.dbeaver.runtime.feature/target/,file:/home/lio/dvp/dbeaver/features/org.jkiss.dbeaver.ui.feature/target/,file:/home/lio/dvp/dbeaver/features/org.jkiss.dbeaver.db.feature/target/,file:/home/lio/dvp/dbeaver/features/org.jkiss.dbeaver.db.ui.feature/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.sample.database/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.ui.tipoftheday/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.cmp.simple/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.model/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.registry/target/,file:/home/lio/dvp/dbeaver/bundles/org.jkiss.utils/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.cmp.simple.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.sql/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.navigator/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.core/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.base/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.connection/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.data/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.hex/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.tasks.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.data.transfer/target/,file:/home/lio/dvp/dbeaver/product/appstore/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.data.gis/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.data.gis.view/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.model.sql/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.data.transfer.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.db2/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.mysql/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.mssql/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.oracle/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.postgresql/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.generic/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.athena/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.bigquery/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.clickhouse/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.db2.zos/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.db2.i/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.derby/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.elasticsearch/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.exasol/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.firebird/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.greenplum/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.h2/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.h2gis/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.hana/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.hive/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.hsqldb/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.informix/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.kognitio/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.netezza/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.opendistro/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.phoenix/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.snowflake.core/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.spanner/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.sqlite/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.teradata/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.vertica/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.ocient/target/,file:/home/lio/dvp/dbeaver/bundles/org.jkiss.wmi/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.mysql.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.db2.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.exasol.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.mssql.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.oracle.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.postgresql.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.generic.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.vertica.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.athena.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.bigquery.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.spanner.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.snowflake.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.sqlite.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.wmi/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.acl/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.session/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.erd/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.ui.locks/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.format.sqlworkbenchj/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.import_config/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.tasks.native/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.tasks.native.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.ui.colortheme/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.app.standalone/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.net.ssh/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.net.ssh.jsch/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.net.ssh.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.slf4j/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.tasks.sql.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.charts/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.dashboard/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.content/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.entity/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.image/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.json/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.xml/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.search/target/,file:/home/lio/.m2/repository/, -installIU, org.jkiss.dbeaver.core.appstore, -destination, /home/lio/dvp/dbeaver/product/appstore/target/products/org.jkiss.dbeaver.core.appstore/linux/gtk/aarch64, -profile, DefaultProfile, -profileProperties, org.eclipse.update.install.features=true, -roaming, -p2.os, linux, -p2.ws, gtk, -p2.arch, aarch64]. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :appstore

**Describe the solution you'd like**
I'm looking forward to Dbeaver running on arm64 architecture.

**Describe alternatives you've considered**
I suggest that since the development of Dbeaver uses java language, it should be able to run on different hardware architectures such as x86_64, arm64 and mips64el.

**Comment by uslss** [comment, marked as answer]
duplicate of #8525

**Comment by SubOptimal** [comment]
Supported plattforms of the Equinox framework. (for a list I checked here https://repo1.maven.org/maven2/org/eclipse/platform/)

Adding following to the `pom.xml` worked for me

```
<environment>
    <os>linux</os>
    <ws>gtk</ws>
    <arch>aarch64</arch>
</environment>
```
`mvn compile package` produces following targets
```
product/standalone/target/products/org.jkiss.dbeaver.core.product/linux/gtk/aarch64
product/standalone/target/products/org.jkiss.dbeaver.core.product/linux/gtk/x86_64
product/standalone/target/products/org.jkiss.dbeaver.core.product/macosx/cocoa/x86_64
product/standalone/target/products/org.jkiss.dbeaver.core.product/win32/win32/x86_64
```

**Reply by rocky91-shi** [reply]
in this way Arm64 platform binary files can be compiled , but can they run normally?

**Comment by rocky91-shi** [comment]
Excuse me,

I also meet the same problem，What is the result of your arm dbeaver compilation?

**Reply by rocky91-shi** [reply]
> **Is your feature request related to a problem? Please describe.**
> 
> By default, Dbeaver compiles different versions for windows, linux and MacOS. They are all based on x86_64 architecture. When I add aarch64 architecture support to the file pom.xml, the compilation will report an error.
> 
> [INFO] org.jkiss.dbeaver.ext.office.feature 1.1.72-SNAPSHOT SUCCESS [ 0.062 s]
> [INFO] org.jkiss.dbeaver.ext.ui.svg.feature 1.0.71-SNAPSHOT SUCCESS [ 0.058 s]
> [INFO] features 1.0.0-SNAPSHOT ............................ SUCCESS [ 0.005 s]
> [INFO] org.jkiss.dbeaver.ext.test 1.0.139-SNAPSHOT ........ SUCCESS [ 0.037 s]
> [INFO] tests 1.0.0-SNAPSHOT ............................... SUCCESS [ 0.005 s]
> [INFO] DBeaver CE AppStore 7.1.1-SNAPSHOT ................. FAILURE [ 13.705 s]
> [INFO] DBeaver CE Standalone 7.1.1-SNAPSHOT ............... SKIPPED
> [INFO] DBeaver P2 Repositories 1.0.0-SNAPSHOT ............. SKIPPED
> [INFO] DBeaver CE Repository 1.0.0-SNAPSHOT ............... SKIPPED
> [INFO] DBeaver - Debug support 1.0.0-SNAPSHOT ............. SKIPPED
> [INFO] DBeaver - Office support 1.0.0-SNAPSHOT ............ SKIPPED
> [INFO] DBeaver - SVG support 1.0.0-SNAPSHOT ............... SKIPPED
> [INFO] DBeaver - SSHJ support 1.0.0-SNAPSHOT .............. SKIPPED
> [INFO] DBeaver - Git support 1.0.0-SNAPSHOT ............... SKIPPED
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time: 01:26 min
> [INFO] Finished at: 2020-06-24T10:41:58+08:00
> [INFO] ------------------------------------------------------------------------
> 
> [ERROR] Failed to execute goal org.eclipse.tycho:tycho-p2-director-plugin:1.7.0:materialize-products (materialize-products) on project appstore: Installation of product org.jkiss.dbeaver.core.appstore for environment linux/gtk/aarch64 failed: Call to p2 director application failed with exit code 13. Program arguments were: [-metadataRepository, file:/home/lio/dvp/dbeaver/product/appstore/target/,file:/home/lio/dvp/dbeaver/product/appstore/target/targetPlatformRepository/, -artifactRepository, file:/home/lio/dvp/dbeaver/product/appstore/target/,file:/resolution-context-artifacts@%252Fhome%252Flio%252Fdvp%252Fdbeaver%252Fproduct%252Fappstore,file:/home/lio/dvp/dbeaver/features/org.jkiss.dbeaver.appstore.feature/target/,file:/home/lio/dvp/dbeaver/features/org.jkiss.dbeaver.rcp.feature/target/,file:/home/lio/dvp/dbeaver/features/org.jkiss.dbeaver.runtime.feature/target/,file:/home/lio/dvp/dbeaver/features/org.jkiss.dbeaver.ui.feature/target/,file:/home/lio/dvp/dbeaver/features/org.jkiss.dbeaver.db.feature/target/,file:/home/lio/dvp/dbeaver/features/org.jkiss.dbeaver.db.ui.feature/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.sample.database/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.ui.tipoftheday/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.cmp.simple/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.model/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.registry/target/,file:/home/lio/dvp/dbeaver/bundles/org.jkiss.utils/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.cmp.simple.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.sql/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.navigator/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.core/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.base/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.connection/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.data/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.hex/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.tasks.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.data.transfer/target/,file:/home/lio/dvp/dbeaver/product/appstore/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.data.gis/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.data.gis.view/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.model.sql/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.data.transfer.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.db2/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.mysql/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.mssql/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.oracle/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.postgresql/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.generic/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.athena/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.bigquery/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.clickhouse/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.db2.zos/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.db2.i/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.derby/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.elasticsearch/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.exasol/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.firebird/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.greenplum/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.h2/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.h2gis/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.hana/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.hive/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.hsqldb/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.informix/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.kognitio/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.netezza/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.opendistro/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.phoenix/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.snowflake.core/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.spanner/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.sqlite/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.teradata/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.vertica/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.ocient/target/,file:/home/lio/dvp/dbeaver/bundles/org.jkiss.wmi/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.mysql.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.db2.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.exasol.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.mssql.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.oracle.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.postgresql.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.generic.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.vertica.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.athena.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.bigquery.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.spanner.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.snowflake.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.sqlite.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.wmi/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.acl/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.session/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.erd/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.ui.locks/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.format.sqlworkbenchj/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.import_config/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.tasks.native/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.tasks.native.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ext.ui.colortheme/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.app.standalone/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.net.ssh/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.net.ssh.jsch/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.net.ssh.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.slf4j/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.tasks.sql.ui/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.charts/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.dashboard/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.content/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.entity/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.image/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.json/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.editors.xml/target/,file:/home/lio/dvp/dbeaver/plugins/org.jkiss.dbeaver.ui.search/target/,file:/home/lio/.m2/repository/, -installIU, org.jkiss.dbeaver.core.appstore, -destination, /home/lio/dvp/dbeaver/product/appstore/target/products/org.jkiss.dbeaver.core.appstore/linux/gtk/aarch64, -profile, DefaultProfile, -profileProperties, org.eclipse.update.install.features=true, -roaming, -p2.os, linux, -p2.ws, gtk, -p2.arch, aarch64]. -> [Help 1]
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR]
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
> [ERROR]
> [ERROR] After correcting the problems, you can resume the build with the command
> [ERROR] mvn -rf :appstore
> 
> **Describe the solution you'd like**
> I'm looking forward to Dbeaver running on arm64 architecture.
> 
> **Describe alternatives you've considered**
> I suggest that since the development of Dbeaver uses java language, it should be able to run on different hardware architectures such as x86_64, arm64 and mips64el.

Excuse me,

I also meet the same problem，What is the result of your arm dbeaver compilation?

**Comment by fanyify** [comment]
Try Dbeaver 7.3.5 I have built out an arrch64 version successfully. Also, check your build environment it needs OpenJDK 11 or above.

**Reply by rocky91-shi** [reply]
> Try Dbeaver 7.3.5 I have built out an arrch64 version successfully. Also, check your build environment it needs OpenJDK 11 or above.
Is it possible to build an arch64 version with OpenJdk8 ?

**Reply by fanyify** [reply]
No, it requires eclipse 2020-12 which some plugins need at least OpenJDK 11.

**Reply by rocky91-shi** [reply]
> No, it requires eclipse 2020-12 which some plugins need at least OpenJDK 11.

Ok, got it, thank you！
